### PR TITLE
[ricos-styles, text-color] did update theme/doc style, add getColors to theme, fix text/highlight color resolver that will consider theme when selection is collapsed

### PR DESCRIPTION
**Pull request by maimonav**

### CODE REVIEW

File: packages/ricos-styles/web/src/models/styles.ts [hunk 1]
```diff
@@ -32,6 +32,13 @@ export interface TextualTheme {
    * @memberof TextualTheme
    */
   toStyleTag(): ReactElement;
+  /**
+   * Returns palette colors for the current theme
+   *
+   * @returns  {PaletteColors}
+   * @memberof TextualTheme
+   */
+  getColors(): PaletteColors | undefined;
```
Comment (lxgreen) [inline packages/ricos-styles/web/src/models/styles.ts:41]: why `| undefined`? 
can we have default palette colors, like color/background that you defined in default theme?  every `undefined` makes our types a bit weaker

File: packages/ricos-editor/web/src/tiptap/RicosStyles.tsx [hunk 2]
```diff
@@ -39,6 +49,17 @@ const RicosStylesRenderer: FC<{
     };
   }, []);
 
+  useEffect(() => {
```
Comment (lxgreen) [inline packages/ricos-editor/web/src/tiptap/RicosStyles.tsx:52]: didn't we remove the `usePrevious` yesterday?

**Review by lxgreen — APPROVED** [review]
nice

File: packages/ricos-styles/web/src/models/styles.ts (hunk 1 above)
Comment (maimonav) [reply packages/ricos-styles/web/src/models/styles.ts:41]: Thanks, fixed this in [here](https://github.com/wix/ricos/pull/4544)

**Pull request merged**